doc/design/katib_on_sqlflow.md

```diff
@@ -64,28 +65,59 @@ We need to develop a new codegen, `codegen_couler.go`, for SQLFlow.  `codegen_co
 
 ### The Integration via Couler
 
-SQLFlow parses each SQL program into an IR, which is a list of statement IRs.  The `codegen_couler.go` converts the IR into a Couler program.   We need to add a Couler functions `couler.katib.train` for the calling by the generated Couler program.
+SQLFlow parses each SQL program into an IR, which is a list of statement IRs.  The `codegen_couler.go` converts the IR into a Couler program.   We need to add a Couler functions `couler.sqlflow.train` for the calling by the generated Couler program.
 
 Consider the following example program.
 
 ```sql
 SELECT * FROM a, b WHERE a.id = b.id INTO c;
-SELECT * FROM c TO TRAIN model_def WITH objective=multi:softmax, eta=1 LABEL class INTO my_xgb_model;
+SELECT * FROM c TO TRAIN data_scientist/xgboost:v0.5/xgboost.gbtree 
+    WITH objective=multi:softmax, eta=0.1, validation_dataset="select * from d;" 
+    INTO my_xgb_model;
 ```
 
 The `codegen_couler.go` might generate the following Couler program.
 
 ```python
-couler.maxcompute.run("""SELECT * FROM a, b WHERE a.id = b.id INTO c;
-                         SELECT * FROM c INTO temp""")
-couler.maxcompute.export(table="temp", file="/hdfs/temp")
-couler.katib.train(model_def, data="/hdfs/temp")
+couler.maxcompute.run("""SELECT * FROM a, b WHERE a.id = b.id INTO c;""")
+couler.sqlflow.train(model="xgboost.gbtree", hyperparameters={"objective": "multi:softmax", 
+    "eta": 0.1},  image="data_scientist/xgboost:v0.5",
+    sql="select * from c to train data_scientist/xgboost:v0.5/xgboost.gbtree ... ",
+    datasource="mysql://..." )
 ```
 
-## `couler.katib.train(...)`
+## `couler.sqlflow.train(...)`
```

Comment:
For katib in couler, it may also serve other applications. So we design a unique API for each application since their parameters may be quite different.